### PR TITLE
[NFC] Remove SkipData argument from getAndSetDebugAbbrevOffsetAndSkip

### PR DESCRIPTION
**Pull request by rastogishubham**
When looking at the code in MCCASObjectV1.cpp, the function `getAndSetDebugAbbrevOffsetAndSkip`, is only called in one place, and the argument `SkipData` is always set to its default value of true, it seems like that argument is never used.

I even tried to build clang with `-fcas-backend-mode=verify` with a clang with this change, and it had no issues

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test llvm

File: llvm/lib/MCCAS/MCCASObjectV1.cpp [hunk 1]
```diff
@@ -1681,13 +1682,10 @@ static Expected<CUInfo> getAndSetDebugAbbrevOffsetAndSkip(
       return std::move(E);
   }
 
-  // Do not skip the CU length when materializing the debug info section, the CU
-  // length will be larger than the CUData length.
-  if (SkipData) {
     Reader.setOffset(AfterSizeOffset);
     if (auto E = Reader.skip(*Size))
       return std::move(E);
```
Comment (felipepiovezan) [inline llvm/lib/MCCAS/MCCASObjectV1.cpp]: Shouldn't the indentation change here? Or is this Github trolling us?

**Review by felipepiovezan — APPROVED** [review]
LGTM! But I am curious about the comment above

File: llvm/lib/MCCAS/MCCASObjectV1.cpp (hunk 1 above)
Comment (rastogishubham) [reply llvm/lib/MCCAS/MCCASObjectV1.cpp]: Ah maybe I forgot to do a `git clang-format` here, I will fix it, sorry!

**Pull request merged**